foreman_ansible_inventory.py

```diff
@@ -124,15 +124,22 @@ def parse_cli_args(self):
                             help='Force refresh of cache by making API requests to foreman (default: False - use cache files)')
         self.args = parser.parse_args()
 
-    def _get_json(self, url):
+    def _get_json(self, url, page = 1, results = []):
         ret = requests.get(url,
                            auth=HTTPBasicAuth(self.foreman_user, self.foreman_pw),
-                           verify=self.foreman_ssl_verify)
+                           verify=self.foreman_ssl_verify,
+                           params={'page': page})
         ret.raise_for_status()
-        return ret.json()
+        if not ret.json().has_key('results'):
+            return ret.json()
+        results = results + ret.json()['results']
+        if len(results) >= ret.json()['total']:
+            return results
+        else:
+            return self._get_json(url, page=page+1, results=results)
```

Comment:
Wouldn't we better not recurse here but use a simple loop? Otherwise we keep references to lots of data for now good reason. And I don't think python optimizes tail recursion ;)